sandpack-react/src/contexts/sandpackContext.tsx

```diff
@@ -519,12 +519,31 @@ export class SandpackProviderClass extends React.PureComponent<
   };
 
   deleteFile = (path: string): void => {
-    this.setState(({ visibleFiles, files }) => {
+    this.setState(({ visibleFiles, files, activeFile }) => {
       const newFiles = { ...files };
       delete newFiles[path];
 
+      const remainingVisibleFiles = visibleFiles.filter(
+        (openPath) => openPath !== path
+      );
+      const deletedLastVisibleFile = remainingVisibleFiles.length === 0;
+
+      if (deletedLastVisibleFile) {
+        const nextFile = Object.keys(files)[Object.keys(files).length - 1];
```

Comment:
@AaronPowell96 I made a few changes to make the next available file instead of the entry file. I believe it's the safest way to go because I might need to consider that any file can be deleted, even the entry file.

What do you think?

Reply:
Okay that's better in my opinion too! Nice work